dist/access_check/MODULARITY.md

```diff
@@ -0,0 +1,44 @@
+# Rules for Modularity
+
+## Definition of a Module
+
+* Whatever resides in a subdirectory under `src/` is considered to be a module.
+
+* The module names are pre-configured as a list of subdirectories under `src/`, with some exclusions like `include`, `checksum`, `os*`, etc. These exclusions are because these directories don’t really contain modular code. If the module name, as deduced by any method, is not in this list, there is no associated module, and the content is exempt from modularity checks.
+
+* Modules can have aliases. `block_cache` refers to `blkcache`, or `connection` is the same as `conn`.
+
+## Modules Content and Visibility Rules
+
+### 1. File Name/Path-Based Rules
+
+1. The default module for the content of a file is defined by these rules:
+   1. If the file is not in the `include` directory, the module is the topmost subdirectory name after `src/`.
+   2. For files in `include/`, the module name is derived from the file name with `.h` and `_inline` stripped.
+
+2. If the file name contains `_private`, everything in that file is considered private by default. Otherwise, everything is public by default.
```

Comment:
Do "private" and "public" here refer to the scope of functions/classes in that file? 1st reading makes me think of file read permission (which I believe is not the case).

Reply:
Further reading down the next few sections gave me more clarity (that it's about identifier scope). It's probably useful to describe "private" and "public" here explicitly.